src/openms_gui/source/VISUAL/Spectrum3DCanvas.cpp

```diff
@@ -192,6 +192,10 @@ namespace OpenMS
   {
 #endif
 
+    // make sure OpenGL already properly initialized
+    QOpenGLContext *ctx = QOpenGLContext::currentContext();
+    if (!ctx || !ctx->isValid()) return;
```

Comment:
how do we deal with failure here? according to the docs:

```
The typical way to handle context loss scenarios in applications is to check via this function whenever makeCurrent() fails and returns false. If this function then returns false, recreate the underlying native OpenGL context by calling create(), call makeCurrent() again and then reinitialize all OpenGL resources.
```
http://doc.qt.io/qt-5/qopenglcontext.html#isValid

Reply:
I don’t think we need to at this point. I think we now just ignore premature update calls.